GSS-Cogs/idpd-api-poc
# 165 log out request

This ticket makes changes to the custom logger so that 'request_id' is added to it. Then at the start of each request handler, request_id is obtained from the env var 'X-Request-ID' and passed into each store method. Each store method now has a new parameter 'request_id: str' and also have at least one logger.info statement that contains the request_id created earlier, regardless of wether request_id has a value or is none.

How to check:
- Check every request handler gathers request_id from env vars and passes request_id into the store methods.
- check every store methods have new parameter 'request_id'
- check every store method has at least one 'logger.info(...., request_id=request_id)'
- (Optional) See if any more logger statements could be placed anywhere a developer would want to know about whats occurred
- check all logger.info statements in store methods make sense/could do with rewording.
- chcek all unit tests in test_custom_logging.py pass.
